scripts/codestyle/run_uncrustify.sh

```diff
@@ -1,103 +1,68 @@
+
+while getopts :qh opt; do
+	case $opt in
+	q)
+		verbose=""
+		;;
+	h)
+		echo "run_uncrustify.sh [options] [<git diff args>]"
+		echo ""
+		echo "Check codestyle of a change. Change generated by bypassing"
+		echo "diff-generating arguments to git-diff"
+		echo ""
+		echo "Options:"
+		echo "-q : show only file names with codestyle issues, suppress"
+		echo "     printing suggested fix"
+		echo "-h : shows this message"
+		exit 1
+		;;
+	?)
+		break
+		;;
 
-while [[ $# -gt 0 ]]
-do
-	key="$1"
-	case $key in
-		-v|--verbose)
-			VERBOSE=true
-			;;
-		-h|--help)
-			echo "run_uncrustify.sh [options] [<SHA1> <SHA2>]"
-			echo ""
-			echo "If SHA is provided to the script, uncrustify checks code style of files that are different in SHA and HEAD. "
-			echo "Otherwise uncommited diff is checked."
-			echo ""
-			echo "Options:"
-			echo "-v|--verbose : shows uncrustify suggestions for code style. When not set, only file names with differences are shown."
-			echo "-h|--help	: shows this message"
-			exit 1
-			;;
-		*)
-			HASHES+=($key)
-			;;
-esac
-shift
+	esac
 done
 
-if [ ${#HASHES[@]} -eq 2 ];
-then 
-	diffargs="${HASHES[0]} ${HASHES[1]}"
-fi
+# the rest will go to git diff directly
+shift $(($OPTIND - 1))
+diffargs="$@"
 
-UNCRUSTIFY_BASE=$(dirname $0)
-cd $UNCRUSTIFY_BASE
-files=$(git diff --name-only $diffargs);
+tmpundiff=$TEMPDIR/undiff
+mkdir -p $tmpundiff
 
-folders_to_format=$(cat folders_to_format.txt)
-echo "Following folders will be processed:"
-echo $folders_to_format;
+git diff --no-prefix $diffargs -- 'src/*.[ch]' | awk -v outpref=$tmpundiff -f $UNCRUSTIFY_BASE/undiff.awk
 
-if [ -s uncrustify_diff.txt ]
-then
-	rm uncrustify_diff.txt
-fi
+result=0
 
-mkdir -p out
+for f in $(cd $tmpundiff; find -type f); do
+	from=$tmpundiff/$f 
+	to=$TEMPDIR/fixed/$f
+	mkdir -p $(dirname $to)
 
-for item in $files ; do
-	if_process_item=false
-	for folder in $folders_to_format ; do
-	if [[ $item == $folder* ]] ; then			
-		if_process_item=true;
-		break;
-	fi
-	done
-	if [ "$if_process_item" = false ] ; then
+	uncrustify --frag -c $UNCRUSTIFY_BASE/uncrustify_cfg.ini -f $from > $to 2>/dev/null
+	if [ ! -s $to ]; then 
 		continue
 	fi
-	origitem=$item
-	item="../../"$item #git shows path starting from repo's root and we're at depth of 2.
-	if ! ([ ${item: -2} == '.c' ] ||	[ ${item: -2} == '.h' ]); then
-		echo $item" is not a .c or .h file"
-		continue;
-	fi
-	dn=$(dirname $origitem)
-	mkdir -p out/$dn
-	uncrustify -f $item -c uncrustify_cfg.ini > out/$origitem
-	if [ -s out/$origitem ]
-	then
-		git diff --no-index -- $item out/$origitem > uncrustify_diff_cur.txt
-		if [ -s uncrustify_diff_cur.txt ]; then
-			echo $origitem >> uncrustify_diff_filenames.txt
-			cat uncrustify_diff_cur.txt >> uncrustify_diff.txt
-		fi
-	fi
-
-done
 
-rm -rf out
 
-if [ -s uncrustify_diff.txt ]
-then
-	echo "Some files don't comply with code style:"
-	if $VERBOSE; then
-		cat uncrustify_diff.txt
+	if [ $verbose ]; then 
+		gitarg="--no-pager diff"
```

Comment:
git-diff should determ color support and seems it thinks there is no one for travis log.
I think I'll force color as you suggest. It will make parsing this script output much harder if possible, but for now it's already really complicated.